contracts/Bounty0xStaking.sol

```diff
@@ -68,6 +70,17 @@ contract Bounty0xStaking is Ownable, Pausable {
         emit Lock(msg.sender, huntersLockAmount[msg.sender], huntersLockDateTime[msg.sender]);
     }
     
+    function depositAndLock(uint _amount) external whenNotPaused {
+        require(_amount != 0);
+        require(ERC20(Bounty0xToken).transferFrom(msg.sender, this, _amount));
+        
+        huntersLockAmount[msg.sender] = SafeMath.add(huntersLockAmount[msg.sender], _amount);
+        huntersLockDateTime[msg.sender] = SafeMath.add(now, lockTime);
+        
+        emit Deposit(msg.sender, _amount, balances[msg.sender]);
```

Comment:
Can you remove the deposit event and just have the `lock`

Reply:
ok